### ISSUE TITLE
Better documentation and handling of `TestFile` in `PackageInfo.g`

### ISSUE THREAD

**Issue by fingolfin** [gap-system/gap#4821]
Currently we don't specify much about what a file specified as `TestFile` in `PackageInfo.g` is supposed to do.

The reality (as encoded in our testing setups in various places) is this:
- it should be either a self-contained `.tst` file (we do `LoadPackage` the package beforehand, though)
- or it should be a `.g` file *which at its end exits GAP after setting an appropriate exit code*. Here, appropriate exit code as usual means: 0 upon success, anything else to indicate a failure.

Package using `TestDirectory` (the vast majority at this point) usually do this by passing the option `exitGAP := true` to it.

Some other packages do it via `FORCE_QUIT_GAP(result)` resp. `QUIT_GAP(result)` (note that `result` also can be a boolean for convenience: `fail` and `false` are mapped to exit code 1, while `true` is mapped to 0).

But this convention is not documented. It either should be documented. Or we should revise how we do things. Simply documenting it at this point is probably easiest.

One could think about ways to revise this system. E.g. one could provide a dedicated function which test files can call to signal the outcome of the tests. The one advantage I can think of for this approach would be that one could then read multiple such testfiles in a single GAP session. However, ...
1. this can also be achieved by having two test files; one which has the actual tests and indicates the outcome; another which reads the first file and then exits GAP based on the outcome
2. in general we want to run testfiles in fresh sessions anyway, because doing other things in the same session beforehand tends to affect tests and result in spurious failures.

At this point, most packages already follow the implicit convention, so I am inclined to just say let's make it official by documenting it, and adjusting any packages where this convention is currently not followed (right now I am aware of `atlasrep` and `ctbllib` by @ThomasBreuer which only set exit codes, but don't quit GAP).

**Comment by ChrisJefferson** [comment]
I agree with trying to best capture what people are already doing, and continuing to allow both `.g` and `.tst` files seems very sensible, as a `.tst` is easier, but packages might always have some special requirement that requires a `.g`.

**Comment by ChrisJefferson** [comment]
If we were going to document things, it might be worth consider adding a `fullTest` option too, which gives "tests which run for a longer period of time" -- we'd probably be willing to run package's full tests occasionally (say up to an hour? or even 2?), at the moment when I want to do that I just look around for a "testall.g" or similar, or just run `TestDirectory` on `tst` and hope it works.
